src/Tabs.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+import core from 'metal';
+import templates from './Tabs.soy';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+/**
+ * UI Component to navigate through tabbed data.
+ */
+class Tabs extends Component {
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {Object} tab
+	 * @param {number} index
+	 */
+	addTab(tab, index) {
+		if (core.isNumber(index)) {
+			this.tabs.splice(index, 0, tab);
+		} else {
+			this.tabs.push(tab);
+		}
+	}
+
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {string} label
+	 * @param {boolean} disabled
+	 * @param {number} index
+	 */
+	addTabByName(label, disabled, index) {
+		if (core.isString(label) && core.isBoolean(disabled)) {
+			let tab = {
+				label: label,
+				disabled: disabled
+			};
+
+			this.addTab(tab, index);
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.tab = this.tabs.findIndex(item => (!item.disabled || !core.isDef(item.disabled)) && !this.disabled);
```

Comment:
This will overwrite `tab` if one is given through the constructor, since it's not checking if it was already set. Even better than manually checking would be to use the `valueFn` configuration option for state properties though, so that you could write something like:

```javascript
Tabs.STATE = {
  tab: {
    // You can also pass a function here, but by passing the name you'll be able to
    // implement it inside your class.
    valueFn: 'findFirstAvailableIndex' 
  }
};
```

Also, this `findIndex` function is nice but it seems like [it doesn't work for IE](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#Browser_compatibility), so unfortunately we can't use it :(